swarm/storage/mru/resource.go

```diff
@@ -773,10 +782,10 @@ func (self *Handler) parseUpdate(chunkdata []byte) (*Signature, uint32, uint32,
 // It is the caller's responsibility to make sure that this data is not stale.
 //
 // A resource update cannot span chunks, and thus has max length 4096
-
 func (self *Handler) UpdateMultihash(ctx context.Context, name string, data []byte) (storage.Address, error) {
-	if isMultihash(data) == 0 {
-		return nil, NewError(ErrNothingToReturn, "Invalid multihash")
+	// \TODO perhaps this check should be in newUpdateChunk()
+	if _, _, err := swarmhash.GetMultihashLength(data); err != nil {
+		return nil, NewError(ErrNothingToReturn, fmt.Sprintf("Invalid multihash: %v", err))
```

Comment:
no need to wrap err in `fmt.Sprintf` - it should be clear where it is coming from and what's wrong with it.